ctk/idp/coverage/NotTested.md

```diff
@@ -0,0 +1,59 @@
+| 8.3.7 - Persistent Identifier | Deployments without such requirements are free to use other kinds of identifiers in their SAML exchanges, but MUST NOT overload this format with persistent but non-opaque values | Extremely hard to test. Testing if a value is non-opaque, IOW you cannot derive anything about the user from the identifier alone, is hard.
+
+### Profiles
+
+| Section | Specification Snippet | Justification
+| --- | --- | --- |
+| 4.1.4.2 - <Response> Usage | If multiple assertions are included, then each assertion's <Subject> element MUST refer to the same principal. It is allowable for the content of the <Subject> elements to differ (e.g. using different <NameID> or alternative <SubjectConfirmation> elements). | Figuring out if the subjects all refer to the same principal cannot be done without actually resolving the subjects to a principal. However, <NameID> is being partially tested by comparing
+
+### Bindings
+
+| Section | Specification Snippet | Justification
+| --- | --- | --- |
+| 3.1.1 - Use of RelayState | If a SAML request message is accompanied by RelayState data, then the SAML responder MUST return its SAML protocol response using a binding that also supports a RelayState mechanism. | The CTK currently supports bindings that support a RelayState mechanism. Once additional bindings are supported, this should be revisited.
+| 3.1.1 - Use of RelayState | Implementations MUST carefully sanitize the URL schemes they permit (for example, disallowing anything but "http" or "https") | Hard to test with little value. Most implementations only deal with http and https anyways. Would need to add a plugin point to attempt and hit every exposed endpoint on seemingly any protocol.
+| 3.4.4 - Security Considerations | It is not a requirement that all possible SAML messages be encodable with a particular set of rules, but the rules MUST clearly indicate which messages or content can or cannot be so encoded. | Impossible to test. Cannot verify that the given encoding rules follow the stated guidelines.
+| 3.4.4 - Message Encoding | A query string parameter named SAMLEncoding is reserved to identify the encoding mechanism used. If this parameter is omitted, then the value is assumed to be "urn:oasis:names:tc:SAML:2.0:bindings:URL-Encoding:DEFLATE". | Currently the CTK only supports DEFLATE encoding.
+| 3.4.4.1 - DEFLATE ENCODING | The following signature algorithms (see \[XMLSig]) and their URI representations MUST be supported with this encoding mechanism: `DSAwithSHA1 – http://www.w3.org/2000/09/xmldsig#dsa-sha1`, `RSAwithSHA1 – http://www.w3.org/2000/09/xmldsig#rsa-sha1` | Impossible to test. Can't know if an encoding we might have never seen before can be signed with `DSAwithSHA1` or `RSAwithSHA1`.
```

Comment:
❓ Could we send signatures signed with `DSAwithSHA1` and `RSAwithSHA1` and see if they can' handle it? We don't really care about extra sig algs. They just need to support those two.